resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientInvoker.java

```diff
@@ -157,13 +160,31 @@ protected Object invokeSync(Object[] args)
    protected ClientInvocation createRequest(Object[] args)
    {
       WebTarget target = this.webTarget;
-      for (int i = 0; i < processors.length; i++)
+
+      Annotation[] pathParams = new Annotation[method.getParameterTypes().length];
+      HashMap<String, Object> pathParamsMap = new HashMap<>();
+
+      for (int i = 0; i < pathParams.length; i++)
       {
-         if (processors != null && processors[i] instanceof WebTargetProcessor)
+         Annotation[] paramAnnotations = method.getParameterAnnotations()[i];
+         for (Annotation annotation : paramAnnotations)
          {
-            WebTargetProcessor processor = (WebTargetProcessor)processors[i];
-            target = processor.build(target, args[i]);
+            if (annotation instanceof PathParam)
+            {
+               pathParamsMap.put(((PathParam) annotation).value(), args[i]);
+               break;
+            }
+         }
+      }
+
+      if (pathParamsMap.size() > 1) {
+         target = target.resolveTemplates(pathParamsMap);
+      }
 
+      for (int i = 0; i < processors.length; i++) {
+         if (processors != null && processors[i] instanceof WebTargetProcessor) {

```

Comment:
If the `processors` can be `null` we need to do the `null` check sooner otherwise `processors.length` will throw an NPE.

Reply:
Thank you for notifying me about this if statement, though this code should not be marked as changed - I may have unknowingly done some format changes and Github decided to highlight them. Similar if statement can be seen on line 185 in the committed file.

Despite that, I believe this check is not necessary, as `processors` is an object created by `ProcessorFactory.createProcessors`which returns an array of objects, no matter if it's empty or not. In my point of view, in order to get null `Processors` , the `ProcessorFactory` should throw an exception itself and thus such case can't happen.

I think that the original idea for the part of this line should be `processors[i]` since it's part of a for loop, but changing the part or completely removing it does not effect tests at all.

Should I fix the code so this check happens before the for loop?

Reply:
Ah right I see. It doesn't matter much, but if you're up for I say we do in this case as it looks odd.

Reply:
No problem, fixed in new commit